connectors/src/resources/zendesk_resources.ts

```diff
@@ -358,23 +410,36 @@ export class ZendeskCategoryResource extends BaseResource<ZendeskCategory> {
 
   async delete(transaction?: Transaction): Promise<Result<undefined, Error>> {
     await this.model.destroy({
-      where: {
-        connectorId: this.connectorId,
-      },
+      where: { connectorId: this.connectorId, categoryId: this.categoryId },
       transaction,
     });
     return new Ok(undefined);
   }
 
   /**
-   * Deletes the category and all associated data (articles).
+   * Deletes all the data relative to a category stored in the db and in the data source (category and articles).
    */
-  // eslint-disable-next-line no-empty-pattern
-  async remove({}: {
+  async deleteWithChildren({
+    dataSourceConfig,
+  }: {
     dataSourceConfig: DataSourceConfig;
-    loggerArgs: Record<string, string | number>;
   }) {
-    // TODO: implement removal for the children first.
+    const articles = await ZendeskCategoryResource.fetchAllArticles({
+      connectorId: this.connectorId,
+      categoryId: this.brandId,
+    });
+    await Promise.all(
+      articles.map((article) =>
+        Promise.all([
+          deleteFromDataSource(
```

Comment:
Ah okay now I understand why you were deleting in a loop one by one. Why did you choose you delete from the datasource here? This function calls front -> core to delete the data source, we usually call it async from an activity. 

Will come to discuss IRL